packages/zowe-explorer/src/extension.ts

```diff
@@ -66,6 +67,16 @@ export async function activate(context: vscode.ExtensionContext): Promise<ZoweEx
     // Determine the runtime framework to support special behavior for Theia
     globals.defineGlobals(preferencesTempPath);
 
+    if (PersistentFilters.getDirectValue("zowe.hideTempFolder") as boolean) {
+        vscode.workspace
+            .getConfiguration("files")
+            .update(
+                "exclude",
+                { [getZoweDir()]: true, [globals.ZOWETEMPFOLDER]: true },
```

Comment:
Testing this with VS Code I see that it adds these settings for me

```json
"files.exclude": {
    "/Users/phaumer/.zowe": true,
    "/Users/phaumer/Dev/Zowe/vscode-extension-for-zowe/packages/zowe-explorer/resources/temp": true
  }
```

However, these settings do not overlap with the workspace path I am using, so they are really unnecessary. Would it make more sense to test if the path of zowe dir and zowe temp actually overlap with the current workspace path and then only if it does to add the setting?

Reply:
While I agree with all the other comments, I do not agree with this one. Zowe Explorer works independently of the vscode workspace. This suggests that we make the code behaviour workspace dependent. I think that would be worse than the current behaviour (which is not ideal either).

This way the behaviour predictable and easy to understand for the users. So I would suggest that we keep it as is.

Reply:
This touches the subject of how we should we properly handle workspace vs. user settings which is a much larger discussion that we will have to have.

Reply:
@VitGottwald can you clarify what you mean by "workspace dependent"? The "files.exclude" setting's purpose is to modify how the workspace is being presented. It has to be specific to the current users workspace. If a user does not use a workspace then then there is no problem. I rather think that we should only fill in values that represent appropriate workspace locations otherwise it would be confusing to the end user or they might think this is a bug.